traincontroller/src/main/java/traincontroller/model/TrainControllerInterface.java

```diff
@@ -0,0 +1,14 @@
+package traincontroller.model;
+
+public interface TrainControllerInterface {
+
+  void setAntennaSignal(Byte[] signal);
+
+  void setBeaconSignal(Byte[] signal);
+
+  void setTrackCircuitSignal(Byte[] signal);
+
```

Comment:
need createInstance method for the CTC

Reply:
Please take a look at TrainControllerFactory now.